src/main/java/org/sagebionetworks/bridge/services/Schedule2Service.java

```diff
@@ -221,6 +253,11 @@ public Schedule2 updateSchedule(Schedule2 schedule) {
                 .orElseThrow(() -> new EntityNotFoundException(Schedule2.class));
         
         CAN_EDIT_SCHEDULES.checkAndThrow(ORG_ID, existing.getOwnerId());
+        
+        return updateScheduleInternal(existing, schedule);
+    }
```

Comment:
Some other options. 1) move the security check to the controller (would involve loading schedule earlier), or possibly CAN_EDIT_SCHEDULES can be made to pass for editors of a schedule via a study.